packages/expo-modules-core/android/src/main/java/expo/modules/kotlin/jni/CppType.kt

```diff
@@ -0,0 +1,24 @@
+package expo.modules.kotlin.jni
+
+private var nextValue = 0
+
+private fun nextValue(): Int {
+  val result = 1 shl nextValue
+  nextValue++
+  return result
+}
+
+/**
+ * Enum that represents a Cpp types. Those types can be obtain via JNI.
+ */
+enum class CppType(private val value: Int) {
+  DOUBLE(nextValue()),
+  BOOLEAN(nextValue()),
+  STRING(nextValue()),
+  JS_OBJECT(nextValue()),
+  JS_VALUE(nextValue()),
+  READABLE_ARRAY(nextValue()),
+  READABLE_MAP(nextValue());
+
+  fun toValue(): Int = value
+}
```

Comment:
Unless you need any magic to call it from C++, then it can be simple:
```suggestion
enum class CppType(val value: Int) {
  DOUBLE(nextValue()),
  BOOLEAN(nextValue()),
  STRING(nextValue()),
  JS_OBJECT(nextValue()),
  JS_VALUE(nextValue()),
  READABLE_ARRAY(nextValue()),
  READABLE_MAP(nextValue());
}
```
And access it via `.value` instead of `.toValue()`